assets/src/data/model/entity-factory/create.js

```diff
@@ -0,0 +1,701 @@
+	} );
+	createPrimaryKeyFieldGetters(
+		instance,
+		keys( primaryKeys )
+	);
+};
+
+/**
+ * Sets the validate type for a field property.
+ * @param {Object} instance
+ * @param {string} fieldName
+ * @param {*} fieldValue
+ */
+const setValidateTypeForField = ( instance, fieldName, fieldValue ) => {
+	instance[ PRIVATE_PROPERTIES.VALIDATE_TYPES ][ fieldName ] =
+		deriveValidateTypeForField( fieldName, fieldValue, instance.schema );
+};
+
+/**
+ *  Populates missing fields and values using defaults provided by schema.  If
+ *  schema doesn't provide a default then this will populate the field with a
+ *  default value that matches the type.
+ *
+ * @param {Object} instance
+ */
+const populateMissingFields = ( instance ) => {
+	if ( ! instance.isNew ) {
+		return;
+	}
+	forEach(
+		getEntityFieldsFromSchema( instance ),
+		( schemaProperties, fieldName ) => {
+			if (
+				! instance[ fieldName ] &&
+				! isPrimaryKeyField( fieldName, instance.schema )
+			) {
+				setInitialEntityFieldsAndValues(
+					instance,
+					fieldName,
+					undefined,
+				);
+			}
+		}
+	);
+};
+
+/**
+ * Returns a plain object of the entity fields and values from this entity
+ * instance prepared for use in an update request.
+ *
+ * @param {Object} instance
+ * @return {Object} Plain object of field:value pairs.
+ */
+const forUpdate = ( instance ) => {
+	return getBaseFieldsAndValuesForPersisting( instance );
+};
+
+/**
+ * Returns a plain object of the entity fields and values from this entity
+ * instance prepared for use in an insert request.
+ *
+ * @param {Object} instance
+ * @return {Object} Plain object of field:value pairs.
+ */
+const forInsert = ( instance ) => {
+	const entityValues = getBaseFieldsAndValuesForPersisting( instance );
+	instance.primaryKeys.forEach( ( primaryKey ) => {
+		entityValues[ primaryKey ] = instance[ primaryKey ];
+	} );
+	return entityValues;
+};
+
+/**
+ * Returns a plain object of the entity fields and values from this entity
+ * instance prepared for use in either an insert or update request.  The type
+ * is automatically derived from the determining whether the entity is "new" or
+ * not.
+ *
+ * @param {Object} instance
+ * @return {Object} Plain object of field:value pairs.
+ */
+const forPersist = ( instance ) => {
+	if ( instance.isNew ) {
+		return forInsert( instance );
+	}
+	return forUpdate( instance );
+};
+
+/**
+ * Creates getters for retrieving the fields and values of the entity instance
+ * for insert or update requests.
+ *
+ * @param {Object} instance
+ */
+export const createPersistingGettersAndSetters = ( instance ) => {
+	createGetter( instance, 'forUpdate', forUpdate );
+	createGetter( instance, 'forInsert', forInsert );
+	createGetter( instance, 'forPersist', forPersist );
+};
+
+/**
+ * Creates initial entity field accessors.
+ *
+ * @param {Object} instance
+ * @param {string} fieldName
+ * @param {*} fieldValue
+ */
+const setInitialEntityFieldsAndValues = (
```

Comment:
The plural name is because there are multiple getters and setter fields that are created from that field and value.

- raw entity getters and setters
- rendered getters

So I think the plural name is a good fit here.